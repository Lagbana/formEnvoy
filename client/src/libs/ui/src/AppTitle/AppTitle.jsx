/* Screen title
=============================== */

import React from 'react';
import { Text, StyleSheet, TextStyle } from 'react-native';
import { Typography, Colors } from 'libs/ui/src/WebButton/node_modules/libs/ui/src/WebLoading/node_modules/colibri/styles';

export interface AppTitleProps {
	color: string;
	style: TextStyle;
}

export const AppTitle: React.FC<AppTitleProps> = ({ children, color = 'text', ...props }) => {
  // styles

  const styles = StyleSheet.create({
    container: {
      ...Typography.titleLarge,
      color: Colors[color],
    },
  });

  return <Text style={[styles.container, props.style]}>{children}</Text>;
};
