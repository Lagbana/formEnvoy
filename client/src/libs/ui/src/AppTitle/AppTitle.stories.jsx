import React from 'react'
import { storiesOf } from '@storybook/react-native'
import { AppTitle } from './AppTitle'

storiesOf('AppTitle', module)
  .add('Blue Title', () => (
    <AppTitle color={'blue'} children={'Hello World!'} style={{}} />
  ))
  .add('Red Title', () => (
    <AppTitle color={'red'} children={'Hello World!'} style={{}} />
  ))
