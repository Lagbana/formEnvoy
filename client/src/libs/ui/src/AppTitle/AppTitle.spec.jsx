import React from 'react'
import { shallow, configure } from 'enzyme'
import toJson from 'enzyme-to-json'
import Adapter from 'enzyme-adapter-react-16'
import { AppTitle } from './AppTitle'

configure({ adapter: new Adapter() })

test('<AppTitle /> renders correctly', () => {
  const component = shallow(<AppTitle color={'blue'} style={{}} />)
  component.unmount()
  expect(toJson(component)).toMatchSnapshot()
})
