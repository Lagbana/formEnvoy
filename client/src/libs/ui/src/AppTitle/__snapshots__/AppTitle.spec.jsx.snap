// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AppTitle /> renders correctly 1`] = `""`;
